src/mpid/ch4/src/ch4r_recvq.h

```diff
@@ -28,9 +28,39 @@ int MPIDIG_recvq_init(void);
 MPL_STATIC_INLINE_PREFIX int MPIDIG_match_posted(int rank, int tag,
                                                  MPIR_Context_id_t context_id, MPIR_Request * req)
 {
-    return (rank == MPIDIG_REQUEST(req, rank) || MPIDIG_REQUEST(req, rank) == MPI_ANY_SOURCE) &&
+    /* FIXME: anysource_partner_request need use atomics */
+    int match;
+    match = (rank == MPIDIG_REQUEST(req, rank) || MPIDIG_REQUEST(req, rank) == MPI_ANY_SOURCE) &&
         (tag == MPIR_TAG_MASK_ERROR_BITS(MPIDIG_REQUEST(req, tag)) ||
          MPIDIG_REQUEST(req, tag) == MPI_ANY_TAG) && context_id == MPIDIG_REQUEST(req, context_id);
+    if (!match) {
+        return 0;
+    } else if (req->kind == MPIR_REQUEST_KIND__ANYSRC_RECV) {
+        MPIR_Request *req2;
```

Comment:
`req2` is a pretty awful variable name. :) How about partner_req so it's more understandable?

Reply:
I was quite obnoxious against current code that calls "partner" and then "partner's partner" :). "partner" offers no more information than `req2`, right? 

Of course I will not insist anything on a local temp name.